presto-docs/src/main/sphinx/develop/worker-protocol.rst

```diff
@@ -125,3 +125,72 @@ upstream workers using buffer number 2.
 
 .. image:: worker-protocol-output-buffers.png
   :width: 600
+
+Failure Handling
+~~~~~~~~~~~~~~~~
+
+Task failures are reported to the coordinator via ``TaskStatus`` and ``TaskInfo``
+updates.
+
+When a task failure is discovered, the coordinator aborts all remaining tasks and
+reports a query failure to the client. When a task failure occurs or an abort
+request is received, all further processing stops, and all remaining task output
+is discarded.
+
+Failed or aborted tasks continue responding to data plane requests as usual to
+prevent cascading failures. Because the output is fully discarded upon failure, all
+following responses are empty. The ``X-Presto-Buffer-Complete`` header is set to
+``false`` to prevent downstream tasks from finishing successfully and producing
+incorrect results.
+
+To the client, these responses are indistinguishable from those of healthy tasks.
+To avoid request bursts, a standard delay before responding with an empty result
+set is applied.
+
+Diagnosing Issues
+~~~~~~~~~~~~~~~~~
+
+HTTP request logging is extremely helpful when diagnosing protocol related problems.
+
+Request logging can be enabled through the ``config.properties`` file.
+
+In Presto:
+
+.. code-block:: none
+
+    http-server.log.enabled=true
+    http-server.log.path=<request_log_file_path>
+
+
+In Prestissimo (logs are written to standard log):
+
+.. code-block:: none
+
+    http-server.enable-access-log=true
+
+By applying simple grep techniques, it is possible to follow a certain protocol interaction.
```

Comment:
```suggestion
Use grep to follow a certain protocol interaction.
```
(This is a big answer for a small change, but I wanted to explain why I suggested this change.) 

Technical documentation should not describe something as "simple" or "easy", because "simple" and "easy" are different for different users. For more context, see Gitlab documentation style guide topic [Building Trust](https://docs.gitlab.com/ee/development/documentation/styleguide/#building-trust), and [Don’t say “simply” in your documentation by Jim Fisher](https://blog.knowledgeowl.com/blog/posts/dont-say-simply-jim-fisher/).

Reply:
Great point. Thank you for explaining.

Reply:
You're welcome! Sometimes I overexplain - like this comment, I suppose - because I've had editors be arbitrary at my writing and I try to be not like them when I'm editing others' work.